API.md

```diff
@@ -0,0 +1,115 @@
+
+So for every `UIViewController` in your app, you would typically have a `Component` subclass to go with it. In our apps, we typically have a `RootComponent`, `MainComponent`, `RideComponent`, and so on. Each *instance* of a `ViewController` subclass typically has it's own instance of the `Component` subclass that it is paired with. **Note:** A lot of the terminology used in conjunction with Needle is inspired by the Dagger DI system on Android.
+
+# Components
+
+Each `Component` is considered to be a `Scope`. The body of the component is relatively simple. It's typically a few variables whose bodies are constructing new objects. Here's an example `Component`
+
+```swift
+import NeedleFoundation
+
+class LoginComponent: Component<LoginDependency> {
+    var tokenProvider: TokenProvider {
+    	return TokenProviderImpl()
+    }
+    
+    var loginViewController: UIViewController {
+       return LoginViewController(tokenProvider: tokenProvider, button: okButton)
+    }
+    
+    var okButton: UIButton {
+       return shared { return UIButton() }
+    }
+}
+```
+**Note:** It's up to you to decide what items make sense on the *DI Graph* and which items can be just local properties in your `ViewController` subclass. Anything that you'd like to mock during a test needs to be passed in (as a protocol) as Swift lacks an `OCMock` like tool.
+
+The `shared` construct in the example is a utility function we provide (in the `Component` base class) that simply returns the same instance every time this `var` is accessed (as opposed to the one above it, which returns a new instance each time).
+
+It's also recommended that you use the component to also construct the `ViewController` that is paired with this component. As you can see in the example above, this allows us to pass in all the dependencies that the `ViewController` needs without the `ViewController` even being aware that you're using a DI system in your project. As noted in the "Benefits of DI" document, it's best to pass in protocols instead of concrete classes or structs.
+
+# Dependencies
+
+If components ended here, they would simply be a container to hold all the "dependencies" of each `ViewController`. This, in itself, is useful as it allows for better unit-tests for your `ViewController` class. Of course, `UIViewController` subclasses are often not easily unit-testable, which is why our RIB architecture (and many others) splits out the "business logic" into a separate class which is easily unit testable.
+
+The real power comes from being able to fetch items from other `Components` as well. 
```

Comment:
Let's be specific here. Instead of saying "other" we should say "ancestors within the same tree".